### PR TITLE
README.ES.md --> 04

### PR DESCRIPTION
**Pull request by zanezhub**
New README.ES.md added to the tutorial 04

### CODE REVIEW

**Comment by andre-richter** [comment]
For the `diff` section, please see https://github.com/rust-embedded/rust-raspberrypi-OS-tutorials/commit/e13edf999226007fc497646d6c6152a59aa296dc

I decided it makes more sense to just reference the main README where the diff is updated regularly.

File: 04_safe_globals/README.ES.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+Una `static QEMU_OUTPUT`, sin embargo, esto no nos permitiría llamar funciones que tomen `&mut self`. Para eso necesitaremos una `static mut`; pero llamar funciones que cambian de estado en una `static mut` es inseguro. El razonamiento del compilador de Rust para esta situación es que ya no puede evitar que múltiples núcleos/hilos cambien los datos al mismo tiempo (es una variable global, así que todos la pueden referenciar desde cualquier lugar. El inspector de préstamos o *borrow checker* no nos puede ayudar en esta situación).
+
+La solución a este problema es hacerle un wrap a la variable global y convertirla en una primitiva de sincronización. En nuestro caso, una variante de una primitiva *MUTual EXclusion*. Se agrega `Mutex` como un rasgo (*trait*) en `synchronization.rs`, y es implementado por el `NullLock` en el mismo archivo. Para hacer que el código se acerque al propósito de la enseñanza, esto hace que dejemos afuera la lógica real dedicada a una arquitectura específica para la protección contra el acceso simultáneo, ya que no lo necesitamos mientras el kernel (núcleo) solo se ejecuta en un solo núcleo con las interrupciones desactivadas.
```
Comment (eldruin) [inline 04_safe_globals/README.ES.md:17]:
```suggestion
La solución a este problema es envolver la variable global en una primitiva de sincronización. En nuestro caso, una variante de una primitiva *MUTual EXclusion*. Se agrega `Mutex` como un rasgo (*trait*) en `synchronization.rs`, y es implementado por el `NullLock` en el mismo archivo. Para hacer que el código sea más entendible, dejaremos fuera la lógica real dedicada a una arquitectura específica para la protección contra el acceso simultáneo, ya que no la necesitamos mientras el kernel (núcleo) solo se ejecute en un solo núcleo con las interrupciones desactivadas.
```

File: 04_safe_globals/README.ES.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+
+El `NullLock` se enfoca en mostrar el concepto principal de Rust de la [mutabilidad interior](https://doc.rust-lang.org/std/cell/index.html). Asegúrate de leerlo. También recomiendo este artículo acerca del [modelo mental preciso para las referencias de tipos en Rust.](https://docs.rs/dtolnay/0.0.6/dtolnay/macro._02__reference_types.html)
+
+Si necesitas comparar el `NullLock` a una implementación real de mutex, puedes revisar las implementaciones en el [*spin crate*](https://github.com/mvdnes/spin-rs) o el [*parking lot crate*](https://github.com/Amanieu/parking_lot).
```
Comment (eldruin) [inline 04_safe_globals/README.ES.md:21]:
```suggestion
Si necesitas comparar el `NullLock` con una implementación real de un *mutex*, puedes revisar las implementaciones en el [*crate spin*](https://github.com/mvdnes/spin-rs) o el [*crate parking lot*](https://github.com/Amanieu/parking_lot).
```

**Review by eldruin — COMMENTED** [review]
Thanks! Just a few corrections/improvements.
Please update the content to only reference the diff as stated by @andre-richter above.

**Comment by andre-richter** [comment]
ping @zanezhub

**Comment by andre-richter** [comment]
Closing due to inactivity.

**Pull request closed without merging**